### ISSUE TITLE
Add requests Client support

### ISSUE THREAD

**Issue by ke-zhang-rd** [NSLS-II/pyolog2#18]
based on discuss https://github.com/NSLS-II/pyolog2/issues/11#issuecomment-635513065

**Comment by danielballan** [comment]
It looks like there is consensus that this is worthwhile, especially given that it wouldn't be difficult.

I suggest we hold off just a little longer to be sure that the original `Client` is stable and designed the way we like it. Once we have two similar-but-different-enough-to-require-two-classes Clients, every change will need to be applied in two places.

**Comment by ke-zhang-rd** [comment]
I wrote some code already in https://github.com/ke-zhang-rd/pyolog2/blob/requests-client/rolog.py but would like to pause as @danielballan mentioned waiting to stable.

I was trying to use vcr record of httpx package in `cassettes` directory to test `requests`. Natively thinking since both are http response, it could just work with only diff in test_olog.py below:
```
- from olog import Client
+ from rolog import Client
```
However, I found `requests` and `httpx` doesn't create identical json-like body by vcrpy.
**In conclusion, we need a new directory maybe `requests_cassettes` for `requests` vcr record**

**Comment by ke-zhang-rd** [comment]
This need to address #19
